src/main/java/com/epam/esm/web/controller/TagController.java

```diff
@@ -2,52 +2,58 @@
 
 import com.epam.esm.model.Tag;
 import com.epam.esm.service.TagService;
-import com.epam.esm.dao.exception.DaoException;
+import com.epam.esm.service.exception.ServiceException;
 import com.epam.esm.service.request.TagRequestBody;
 import com.epam.esm.web.dto.TagDto;
-import com.epam.esm.web.hateoas.TagProcessor;
+import com.epam.esm.web.hateoas.ModelAssembler;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.Page;
 import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
 import org.springframework.hateoas.PagedModel;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.List;
 
 @RestController
 @RequestMapping("/tags")
 public class TagController {
 
     private final TagService tagService;
-    private final TagProcessor tagProcessor;
+    private final ModelAssembler<TagDto> modelAssembler;
 
     @Autowired
-    public TagController(TagService tagService, TagProcessor tagProcessor) {
+    public TagController(TagService tagService, ModelAssembler<TagDto> modelAssembler) {
         this.tagService = tagService;
-        this.tagProcessor = tagProcessor;
+        this.modelAssembler = modelAssembler;
     }
 
     @GetMapping
```

Comment:
Where is method documentation?

Reply:
Javadoc will be added after everything else is finished